src/module/reservation.js

```diff
@@ -0,0 +1,142 @@
+    const inner = document.querySelector('.popup-inner');
+    inner.insertBefore(currentReservations, form);
+    const url2 = `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/FXMctN9lLmEpmOxXkx1x/reservations?item_id=${idMeal}`;
+    const getRes = async () => {
+      const res = await fetch(url2);
+      const info = await res.json();
+      return info;
+    };
+    getRes().then((res) => res.forEach((item) => {
+      if (item) {
+        currentReservations.innerHTML += `
+         <li class = "reservations-list">${item.username} Made a Reservation on ${item.date_start} till ${item.date_end}</li>
+        `;
+      }
+    }));
+    const from = document.querySelector('#makeRes');
+    from.addEventListener('submit', (e) => {
+      e.preventDefault();
+      const name = document.querySelector('#makeRes input[type="name"]').value;
+      const start = document.querySelector('#makeRes input[type="date"]').value;
+      const end = document.querySelector('#makeRes input[type="date"]').value;
+      const url3 = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/FXMctN9lLmEpmOxXkx1x/reservations';
+      const postRes = async () => {
+        const res = await fetch(url3, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            item_id: idMeal,
+            username: name,
+            date_start: start,
+            date_end: end,
+          }),
+        });
+        const info = await res.json();
+        return info;
+      };
+      postRes();
+      const success = document.createElement('div');
+      success.classList.add('successMessage');
+      success.innerHTML = 'Your Reservation Was Made ';
+      const form = document.querySelector('.form-inline');
+      form.appendChild(success);
+      setTimeout(() => {
+        form.removeChild(success);
+        window.location.reload();
+      },
+      3000);
+    });
+    const end = document.querySelector('#end');
+    end.addEventListener('click', () => {
+      document.querySelector('body').removeChild(popup);
+    });
+  });
+
+  const url4 = `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/FXMctN9lLmEpmOxXkx1x/reservations?item_id=${idMeal}`;
+  const getTotalRes = async () => {
+    const res = await fetch(url4);
+    const info = await res.json();
+    return info;
+  };
+  getTotalRes().then((res) => {
+    const total = res.length;
+    const totalReservationsOnMeal = document.createElement('div');
+    totalReservationsOnMeal.classList.add('totalReservationsOnMeal');
+    if (total === undefined) {
+      totalReservationsOnMeal.classList.add('noReservation');
+      totalReservationsOnMeal.innerHTML = '<h3> No Reservations On this Meal </h3>';
+    } else {
+      totalReservationsOnMeal.innerHTML = `
+        <h3>Total Reservations on this Meal : <span class ="Spani">${total}</span></h3>
```

Comment:
- [ ] Consider changing the "Total Reservations on this Meal: #" title to something shorter like "Reservations (#)"